src/components/Form.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { InputStyled, Label } from '../styles/Taqstyles'
+import { FormErrors } from './FormErrors';
+
+export interface FormProps {
+  type: string;
+  setFieldValue:(value:string, type:string, valid:boolean) => (void)
+}
+
+export interface FormState {
+  formErrors: any;
+  content: string;
+  valid: boolean;
+}
+
+export default class Form extends React.Component<FormProps, FormState> {
+  constructor(props: FormProps) {
+    super(props);
+    this.state = {
+      content: '',
+      valid: true,
+      formErrors: { content: '' },
+    }
+  }
+  render() {
+    const type = this.props.type;
+    return (
+      <>
+        <div style={{textAlign:"left"}}>
+        <Label>{this.decideLabel(type)}</Label>
+        </div>
+        <InputStyled
+          name={type}
+          type={type}
+          valid={this.state.valid}
+          onBlur={event => this.validateField(type, event.target.value)}
+        />
+          <FormErrors formErrors={this.state.formErrors} />
+      </>
+    )
+  }
+
+   private decideLabel = (type: string) => {
+    let labelString: string = '';
+    switch (type) {
+      case 'email':
+        labelString = "E-mail: "
+        break;
+      case 'name':
+        labelString = "Nome: "
+        break;
+      case 'password':
+        labelString = "Senha: "
+        break;
+      case 'cpf':
+        labelString = "CPF: "
+        break;
+      case 'birthDate':
+        labelString = "Data de nascimento: "
+        break;
+      case 'role':
+        labelString = "Role: "
+        break;
+      default:
+        break;
+    }
+    return labelString;
+  }
+
+  private updateContent = (e: any) => {
+    const { value } = e.target;
+    this.setState({ content: value })
+  }
+
+  private validateContent = (e: any, type:string) => {
```

Comment:
Você já tem o type nas props, não precisa passar como parâmetro, certo?

`this.props.type`